### ISSUE TITLE
how make upload request

### ISSUE THREAD

**Issue by thg303** [rswag/rswag#53]
I tried this code to document an upload via put method, but it simply breaks. am I doing something wrong?

```path '/api/v1/puzzles/{puzzle_id}' do
     put 'update a puzzle details' do
      consumes: 'multipart/form-data'
      parameter name: 'Authorization', in: :header, type: :string, required: true
      parameter name: 'Content-Type', in: :header, type: :string, required: true
      parameter name: 'puzzle_id', in: :path, type: :string, required: true
      parameter name: 'puzzle[kind]', in: :formData, type: :string, enum: %w(freezed motioning), required: false
      parameter name: 'puzzle[picture]', in: :formData, type: :file, required: false
      response 200, 'successful' do
        let(:puzzle) { create :puzzle }
        let(:Authorization) { "Bearer #{get_token(puzzle.user)}"}
        let(:'Content-Type') { 'multipart/form-data' }
        let(:puzzle_id) { puzzle.id }
        let(:'puzzle[kind]') { 'freezed' }
        let(:'puzzle[picture]') { uploaded_file }
        run_test!
      end
    end
```

**Comment by thg303** [comment]
mm.. looks like :formData is not implemented

**Comment by raydog153** [comment]
Do you have a typo? for the parameter of type file you have in as 'formDate' and not 'formData'. I'm still trying to figure out how to set header and host for the request.

**Comment by thg303** [comment]
@raydog153 oh yeah, that was a typo. but it does not work anyway.
for the header you may use my code, that works for simple queries.

**Comment by domaindrivendev** [comment]
@thg303 - you're right form data not supported. I think it could be trivial enough - fancy making a PR?

**Comment by pifleo** [comment]
Hello,

We had the same problem today and found out it was not implemented. We didn't manage to upload a file with `:body`.

So here is our patch to get a working file upload with `:formData` and `type: :file`:
```
# rswag-specs/lib/rswag/specs/request_factory.rb

module Rswag
  module Specs
    class RequestFactory

      # ...

      def build_body(example)
        if parameters_in(:formData).present?
          build_form_data(example)
        else
          body_parameter = parameters_in(:body).first
          body_parameter.nil? ? '' : example.send(body_parameter[:name]).to_json
        end
      end

      private

      def build_form_data_value(param, example)
        param[:type] == :file ? example.send(param[:name]) : example.send(param[:name]).to_s
      end

      def build_form_data(example)
        Hash[ parameters_in(:formData).map { |p| [ p[:name], build_form_data_value(p, example) ] } ]
      end

      # ...
    end
  end
end
```

**Comment by thg303** [comment]
did it, @domaindrivendev mind to check my PR?

**Comment by domaindrivendev** [comment]
Should be fixed by #54
